### PR TITLE
add rustup to requirements and a note about libhwloc-dev on linux

### PR DESCRIPTION
**Pull request by 44203**
just a couple of readme updates when setting up from scratch on ubuntu

### CODE REVIEW

**Review by en0ma — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -128,6 +129,8 @@ For M1 Macs, here's the following steps needed
 - Step 2: export LIBRARY_PATH=/opt/homebrew/lib
 - Step 3: Follow the steps as per the docs.
 
+On Ubuntu, install `libhwloc-dev`.
```
Comment (gmelodie) [inline README.md:132]: This should be in the same line as `- [hwloc](https://www.open-mpi.org/projects/hwloc/)`

**Review by gmelodie — CHANGES_REQUESTED** [review]

**Comment by gmelodie** [comment]
Also why do we need `rustup`?

**Comment by 44203** [comment]
> Also why do we need `rustup`?

Unsure, but running `make clean all` fails if it isn't installed and asks for it.

**Comment by alvin-reyes** [comment]
Please move use `dev` as the upstream branch for this PR.

**Pull request merged**